### PR TITLE
fix syntax on pin-init rust code example

### PR DESCRIPTION
**Pull request by dswij**

### CODE REVIEW

**Comment by ojeda** [comment]
Cc @y86-dev.

I guess `Self` could used instead in both `new()`s.

**Comment by dswij** [comment]
Yeah, that's fair. I guess it's more Rusty that way.

File: src/The-Safe-Pinned-Initialization-Problem.md [hunk 1]
```diff
@@ -50,7 +50,7 @@ impl ListHead {
     /// # Safety
     ///
     /// Before using this [`ListHead`] the caller has to call [`ListHead::init`].
-    unsafe fn new() -> ListHead {
+    unsafe fn new() -> Self {
         ListHead {
```
Comment (y86-dev) [inline src/The-Safe-Pinned-Initialization-Problem.md:54]: I think then you can also change it on this line and below as well.

**Comment by ojeda** [comment]
LGTM, should I publish it, @y86-dev?

**Comment by y86-dev** [comment]
yup also LGTM

**Pull request merged**